### DISCUSSION TITLE
Angularの情報どこで収集してますか？

### DISCUSSION THREAD

**Discussion by scrpgil** [StartupAngular/discussions#37] [General]

**Comment by scrpgil** [comment]
セッション1

**Reply by scrpgil** [reply]
# 概要
## ブレスト
- UdemyでAngularの教材を買ってやっていた。
- Angular公式のチュートリアル：https://angular.io/tutorial
  - Afeterチュートリアル：https://zenn.dev/lacolaco/books/angular-after-tutorial/viewer/introduction
- twitterでAngular公式やAngular使ってる方をフォローしてる。
  - 気になった時に追加で情報収集している。
- Twitterでフォローしている。エンジニア関係のフォロワーさんが多いので、
- ng-japanのSlackが手堅い。
  - https://community.angular.jp/
  - announceをウォッチしている。
- 転職サイトの求人を見る。どういう技術を求めているか。
- Speekerdeckなどのスライド。
- Angular公式のTour Of Herosをやる。
- すでにあるソースコードを見たりしてキャッチアップする。
- ng-japan on air。月1。
- ng-sake。半年に一回。
- ng-japan meetup。
- 社内に強い人が聞いたりする。
- Angularコミュニティのdiscordがある。https://discord.com/invite/angular

## Twitter。
- lacolacoさん：https://twitter.com/laco2net
- Quramyさん：@quramy
- Angular公式：@ng_japan

## 動画講座
- UdemyのAngularとrxjsを使う講座。オブザーバブルの概念的な話。2つのチュートリアルを見ながら作るもの。元々業務でAngularを使っていたけど、改めて学びなおしで使っている。
  - Udemy、若い子たちに人気ある。自分で調べるだけだと書けてる情報が保管できる。
  - Udemyは日本語もある。
  - https://www.udemy.com/courses/search/?lang=ja&q=Angular+%E6%97%A5%E6%9C%AC%E8%AA%9E&sort=relevance&src=ukw
     - 日本語のコンテンツで522ぐらいある。
     - これがおすすめかも？： https://www.udemy.com/course/angular-ja/
  - rxjsなどの概念も難しい。

## Speekerdeck
- https://speakerdeck.com/okunokentaro/frontend-conference-2019

## 余談
- ディレクトリ構造について話した。
  - Ionicでもsharedはない。
  - Angular、自由度高いから逆に難しい。
  - reactにあるbulletproof-reactというようなプロジェクトがAngularにもあると嬉しい。

**Comment by rettar5** [comment]
ルーム2

**Reply by rettar5** [reply]
## どうですか？
- 収集できてないです
   - Twitterで流れてる情報とかしか見れてない
- Twitterで流れてくる情報見てる
- 公式ドキュメント翻訳して読んでる
    - 日本語化リポジトリの翻訳されてない記事を翻訳しながら読んでる
- 前職だとAngularのプロジェクトなので、みんながアンテナ貼ってた
    - AngularのSlackグループや、Twitterや、(個人名)さんや
    - 年単位でエージェントからプロジェクト増えた減ったなどの情報を聞くことがある
    - カンファレンスのYoutubeがおすすめに上がってきたら見てる
- すでに書かれてるものと、ひたすらググる
    - 公式がload map公開してるので、そういうのを見たりしてる
    - 基本は詰まったときとかに調べてる
    - 新しいやり方を得たりしてる
- ngjapan onair頼み
    - updateするときのmigration guideを見てググってる
    - 古い情報を見ることが多い
    - 一回やり方覚えたら泥臭いやり方でいけることが多い
        - 下手なやり方覚えるよりかは泥臭いやり方のがupdateが楽
    - moduleをstandaloneにしましょうってどうやって情報追えばいい？
        - https://github.com/angular/angular/discussions/43784
            - 14に間に合いそうらしい
        - github見るしかない…？
        - angularのリポジトリをwatchしたけど、めちゃくちゃ流れてきてやめました

## こういうところに情報あるらしいよ？みたいなのある？
- tech feedに情報があるらしいけど使ってる方います？
    - https://techfeed.io
- (Angularじゃないけど)はてなブックマークのRSSをSlackに流してみてる
- こういう勉強会で教えてもらう
- なんだかんだAngular強い人から聞くのが多い
    - こういう強い人どうやって見つけてます？
        - ぐぐってよく出てきた人…かな……
        - アドベントカレンダーで「一回で理解できないことを書いてる人」は強い
            - 内部構造知ってる人とかだいたい強い

## このテーマむずいね
- 最新の情報か、今直面してる問題か、歴史か
    - 最新の情報なら勉強会
    - 直面してる問題ならぐぐるよね
    - 歴史は有識者に聞く
- 枯れた情報としてAngularのドキュメントは見てる

## 情報収集するのに困ったテーマってある？
- Angular animationドキュメントなさすぎ問題
- rxjsでハマった
    - ドキュメント簡素で、2日ぐらい泣きながらソースコード読んだ
    - どのオペレーターが正解なのかわからない
    - これか？って書くとみんなduplicatedがついてる
- Angular Schematics
    - https://angular.jp/guide/schematics
    - ngコマンドでng moduleにaddしたりなどしてるがドキュメントがない
    - 開発者はどうやって管理してるの？
    - Qiitaとかだと簡単な使い方が乗ってるが、ガッツリ使ってるケースがなかなか検索できない
    - ngrxのstoreも同じ
    - lightな例はあるが、実際使うときに四苦八苦してる
- 規模が大きくなると情報出しにくいのでは…？
    - 四苦八苦してほしいわけじゃないが社外に出せない情報が……
- Angularで解決しない問題
    - SSRするんだったらAngular以外を選ぶ風潮があり、情報がぜんぜん出てこない
- 人づてに聞くのが一番いい
    - closedな場なら安心して聞ける
